packages/back-end/src/util/organization.util.ts

```diff
@@ -82,13 +152,45 @@ export function getUserPermissions(
     projects: {},
   };
 
+  // Build the user's user-level project permissions
   memberInfo?.projectRoles?.forEach((projectRole: ProjectMemberRole) => {
     userPermissions.projects[projectRole.project] = {
       limitAccessByEnvironment: projectRole.limitAccessByEnvironment || false,
       environments: projectRole.environments || [],
       permissions: roleToPermissionMap(projectRole.role, org),
     };
   });
+
+  // If the member's global role is admin, they already have all permissions
+  if (memberInfo?.role) {
+    const teamsUserIsOn = memberInfo?.teams || [];
+    for (const team of teamsUserIsOn) {
+      const teamData = await findTeamById(team, org.id);
+      if (teamData) {
+        mergePermissions(
+          userPermissions.global,
+          memberInfo.role,
+          teamData,
+          org
+        );
+        if (teamData?.projectRoles && teamData?.projectRoles.length > 0) {
+          for (const teamProject of teamData.projectRoles) {
+            const existingProjectData = memberInfo.projectRoles?.find(
+              (project) => project.project === teamProject.project
+            );
+
+            mergePermissions(
+              userPermissions.projects[teamProject.project],
+              existingProjectData?.role,
+              teamProject,
+              org
+            );
+          }
+        }
```

Comment:
Consider doing this within `mergePermissions` 
Seems like it might make more sense for that function to handle all permission merging logic in case we ever call this somewhere else and you can just call it recursively if there are project roles

Reply:
I abstracted this logic out 1 level with a new method `mergeUserWithTeamPermissions` that encapsulates this logic so we have a single method we can call and pass in the userPermissions, memberInfo, and org, and merge it all together.